components/brave_extension/extension/brave_extension/content_cosmetic.ts

```diff
@@ -584,6 +585,333 @@ const scheduleQueuePump = (hide1pContent: boolean, generichide: boolean) => {
+        }
+        break
+      }
+      case 'iframe': {
+        const url = elem.getAttribute('src')?.trim()
+        if (url !== undefined && url.length > 0) {
+          attributes.push({
+            attr: 'src',
+            value: url.slice(0, 256)
+          })
+        }
+        break
+      }
+      case 'img': {
+        let data = elem.getAttribute('src')?.trim()
+        if (data !== undefined && data.length > 0) {
+          // https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs
+          if (data.startsWith('data:')) {
+            data = data.split(',')[1].slice(0, 256)
+          }
+        }
+        if (data === undefined || data.length === 0) {
+          let alttext = elem.getAttribute('alt')?.trim()
+          if (alttext !== undefined && alttext.length > 0) {
+            attributes.push({
+              attr: 'alt',
+              value: alttext
+            })
+          }
+        } else {
+          attributes.push({
+            attr: 'src',
+            value: data
+          })
+        }
+        break
+      }
+      default: { break }
+    }
+    if (attributes.length > 0) {
+      builder.addRule({
+        type: Selector.Attributes,
+        value: attributes
+      })
+    }
+  }
+
+  const querySelectorNoExcept = (node: Element | null, selector: string): Element[] => {
+    if (node !== null) {
+      try {
+        let r = node.querySelectorAll(selector)
+        return Array.from(r)
+      } catch (e) { /* Deliberately left empty */ }
+    }
+    return []
+  }
+
+  if (builder.size() === 0 || querySelectorNoExcept(elem.parentElement, builder.toString()).length > 1) {
+    builder.addTag(tag)
+    if (querySelectorNoExcept(elem.parentElement, builder.toString()).length > 1) {
+      let index = 1
+      let sibling: Element | null = elem.previousElementSibling
+      while (sibling !== null) {
+        if (sibling.localName === tag) { index++ }
+        sibling = sibling.previousElementSibling
+      }
+      builder.addRule({
+        type: Selector.NthOfType,
+        value: index
+      })
+    }
+  }
+  return builder
+}
+
+const elementPickerOnKeydown = (event: KeyboardEvent): void => {
+  if (event.key === 'Escape') {
+    event.stopPropagation()
+    event.preventDefault()
+    // tslint:disable-next-line:no-use-before-declare
+    quitElementPicker()
+  }
+}
+
+const elementPickerViewportChanged = () => {
+  // tslint:disable-next-line:no-use-before-declare
+  recalculateAndSendTargets(targetedElems)
+}
+
+const quitElementPicker = () => {
+  if (pickerFrame !== null) {
+    document.documentElement.removeChild(pickerFrame)
+  }
+
+  // Tear down element picker listeners
+  document.removeEventListener('keydown', elementPickerOnKeydown, true)
+  document.removeEventListener('resize', elementPickerViewportChanged)
+  document.removeEventListener('scroll', elementPickerViewportChanged)
+}
+
+const launchElementPicker = () => {
+  pickerFrame = document.createElement('iframe')
```

Comment:
have you checked if this works on pages with a `frame-src` content security policy?

Reply:
I didn't even know this CSP existed. Just tested it out and I added `Content-Security-Policy: frame-src https://example.com/` to [my site](http://segfault.fun) config. The frame for brave.com gets broken by the CSP but I can still load the element picker. Seems like it works.